### DISCUSSION TITLE
自定义页眉内容

### DISCUSSION THREAD

**Discussion by tiejundong** [nju-lug/NJUThesis#85] [Q&A] [answered]
大佬你好，目前我院系需要自定义一下正文页眉的内容（研究生版）。我在\mainmatter后用\pagestyle加了下不过似乎没有效果，请问在有什么快捷的方法能加上吗。

P.S. win10上不知怎么的不能正确引用bib文件，显示的是我\cite内的字符内容

邻近毕业季，大佬能在百忙之中处理下就太好了

**Comment by atxy-blip** [comment, marked as answer]
> 自定义一下正文页眉的内容

使用 `fancyhdr` 包提供的设置方法即可。可惜你没说具体改成什么样，这里就给个简单的例子吧：

```TeX
\documentclass[degree=mg]{njuthesis}
\usepackage{zhlipsum}
\begin{document}
\mainmatter
\fancyhead{} % 清空已有页眉
\fancyhead[C]{\small\kaishu 哈哈}
\chapter{占位}
\zhlipsum[1-10]
\end{document}
```

> 显示的是我\cite内的字符内容

需要用 `biber` 命令生成参考文献信息。如果你用的是 TeXstudio 程序，请检查构建设置中的参考文献工具。

另：一个主题内请不要打包发布不同提问

**Reply by tiejundong** [reply]
谢谢大佬，之前我自己用fancyhead[C]没用原来是第X章当页不会显示页眉，而我又没写到第二页，用大佬给的demo发现第二页开始就有了。如果第X章当页有就完美了

**Reply by atxy-blip** [reply]
`\chapter` 当页默认使用 `plain` 样式，在 `\mainmatter` 后加一句 `\fancypagestyle{plain}{}` 清空设置就行。

**Reply by tiejundong** [reply]
感谢大佬了，拯救苍生
